pkg/log/zap/zap.go

```diff
@@ -255,6 +277,14 @@ func (o *Options) BindFlags(fs *flag.FlagSet) {
 func UseFlagOptions(in *Options) Opts {
 	return func(o *Options) {
 		*o = *in
+		if o.Encoder != nil && o.TimeEncoder != nil {
+			ecfs = append(ecfs, withTimeEncoding(o.TimeEncoder))
+			if encVal.value == "json" {
+				o.Encoder = newJSONEncoder(ecfs)
+			} else {
+				o.Encoder = newConsoleEncoder(ecfs)
+			}
+		}
```

Comment:
Is this block of code handled by the added logic in the `addDefaults()` function?

Reply:
@joelanford as I understand `addDefaults()`, will add the timeEncoder to the default Encoder value. Above block of code is for when `Encoder` is being set from flags.